src/react-components/room/useResizeViewport.js

```diff
@@ -8,35 +8,28 @@ import {
   getMaxResolutionHeight
 } from "../../utils/screen-orientation-utils";
 
-// Modified from AFrame
-function getRenderResolution(canvasRect, maxResolution, isVR) {
-  const pixelRatio = window.devicePixelRatio;
-
-  if (!maxResolution || isVR || (maxResolution.width === -1 && maxResolution.height === -1)) {
-    return canvasRect;
-  }
-
-  if (canvasRect.width * pixelRatio < maxResolution.width && canvasRect.height * pixelRatio < maxResolution.height) {
+function calculateRendererSize(canvasRect, maxResolution, isVR) {
+  if (isVR) {
     return canvasRect;
   }
 
-  const aspectRatio = canvasRect.width / canvasRect.height;
+  // canvasRect values are CSS pixels based while
+  // maxResolution values are physical pixels based (CSS pixels * pixel ratio).
+  // Convert maxResolution values to CSS pixels based.
+  const pixelRatio = window.devicePixelRatio;
+  const maxWidth = maxResolution.width / pixelRatio;
+  const maxHeight = maxResolution.height / pixelRatio;
 
-  if (canvasRect.width * pixelRatio > maxResolution.width && maxResolution.width !== -1) {
-    return {
-      width: Math.round(maxResolution.width / pixelRatio),
-      height: Math.round(maxResolution.width / aspectRatio / pixelRatio)
-    };
+  if (canvasRect.width <= maxWidth && canvasRect.height <= maxHeight) {
+    return canvasRect;
```

Comment:
Minor since we only ever read `width` and `height` from the result of this function anyway, but this function end sup having multiple return types. Here we are returning the passed in `DOMRectReadOnly`, but elsewhere we are returning a new object `{width, height}`.

Reply:
Honestly I didn't really like that multiple return types. But the current implementation does it, too. I just didn't change. We may be able to clean up later.